Sources/Target/Linux/Process.cpp

```diff
@@ -339,6 +339,7 @@ ErrorCode Process::getMemoryRegionInfo(Address const &address,
     uint64_t offset;
     unsigned int devMinor, devMajor;
     uint64_t inode;
+    char name[PATH_MAX];
```

Comment:
I think by having an array of size `PATH_MAX`, you are *mostly* good, but there are two things I'd do differently:
* `char name[]` should be of size `PATH_MAX + 1` to account for the terminating null character, in case the path is actually of size `PATH_MAX`;
* you should specify the "field size" in your call to `::sscanf()` so you're guaranteed to never go over the size of your array.

Specifying the size in the format string seems to be a PITA because you have to include the numeric value *in* the format string itself, so you have to double-expand `PATH_MAX`...

Looks like there's also a GNU extension that allows you to use the `m` modifier to let sscanf allocate the memory for you, but then you have to `free()` it yourself, and it's not portable:

> • An optional 'm' character. This is used with string conversions (%s, %c, %[), and relieves the caller of the need to allocate a corresponding buffer to hold the input: instead, scanf() allocates a buffer of sufficient size, and assigns the address of this buffer to the corresponding pointer argument, which should be a pointer to a char * variable (this variable does not need to be initialized before the call). The caller should subsequently free(3) this buffer when it is no longer required.

😢